mods/ra/maps/allies-11a/allies11a-AI.lua

```diff
@@ -0,0 +1,579 @@
+	if BadGuyCYard.IsDead or BadGuyCYard.Owner ~= BadGuy then
+		return
+	end
+	for i,v in ipairs(BaseBuildings) do
+		if not v.exists then
+			BuildBuilding(v)
+			return
+		end
+	end
+	Trigger.AfterDelay(DateTime.Seconds(5), BuildBase)
+end
+
+BuildBuilding = function(building)
+	Trigger.AfterDelay(Actor.BuildTime(building.type), function()
+		if BadGuyCYard.IsDead or BadGuyCYard.Owner ~= BadGuy then
+			return
+		end
+		local actor = Actor.Create(building.type, true, { Owner = BadGuy, Location = BadGuyCYard.Location + building.pos })
+		BadGuy.Cash = BadGuy.Cash - building.cost
+
+		building.exists = true
+		if building.type == "barr" then
+			Trigger.AfterDelay(DateTime.Seconds(40), ProduceBadGuyPeriodic1Units)
+		elseif building.type == "weap" then
+			Trigger.AfterDelay(DateTime.Seconds(120), ProduceBadGuyPeriodic3Units)
+		end
+		Trigger.OnKilled(actor, function() 
+			building.exists = false 			
+		end)
+		Trigger.OnDamaged(actor, function(building)
+			if building.Owner == BadGuy and building.Health < building.MaxHealth * 3/4 then
+				building.StartBuildingRepairs()
+			end
+		end)
+		Trigger.AfterDelay(DateTime.Seconds(5), BuildBase)
+	end)
+end
+
+ActivateAI = function()
+	local difficulty = Map.LobbyOption("difficulty")
+	ProductionInterval = ProductionInterval[difficulty]
+	blk1Units = blk1Units[difficulty]
+	blk2Units = blk2Units[difficulty]
+	blk3Units = blk3Units[difficulty]
+	blk4Units = blk4Units[difficulty]
+	blk5Units = blk5Units[difficulty]
+	periodic1Units = periodic1Units[difficulty]
+	periodic2Units = periodic2Units[difficulty]
+	periodic3Units = periodic3Units[difficulty]
+	periodic4Units = periodic4Units[difficulty]
+	badguyPeriodic1Units = badguyPeriodic1Units[difficulty]
+	badguyPeriodic2Units = badguyPeriodic2Units[difficulty]
+	badguyPeriodic3Units = badguyPeriodic3Units[difficulty]
+	badguyPeriodic4Units = badguyPeriodic4Units[difficulty]
+	local buildings = Utils.Where(Map.ActorsInWorld, function(self) return self.Owner == USSR and self.HasProperty("StartBuildingRepairs") end)
+	Utils.Do(buildings, function(actor)
+		Trigger.OnDamaged(actor, function(building)
+			if building.Owner == USSR and building.Health < building.MaxHealth * 3/4 then
+				building.StartBuildingRepairs()
+			end
+		end)
+	end)
+	Trigger.AfterDelay(DateTime.Minutes(5), ProduceSovietBlk1Vehicle)
+	Trigger.AfterDelay(DateTime.Seconds(210), function()
+		SendRenAirstrike(airGroup1, airGroup1Route)
+	end)
+	Trigger.AfterDelay(DateTime.Seconds(288), function()
+		SendRenAirstrike(airGroup2, airGroup2Route)
+	end)
+	Trigger.AfterDelay(DateTime.Seconds(432), function()
+		SendRenAirstrike(airGroup3, airGroup3Route)
+	end)
+	Trigger.AfterDelay(DateTime.Seconds(528), function()
+		SendRenAirstrike(airGroup4, airGroup4Route)
+	end)
+	Trigger.AfterDelay(DateTime.Seconds(720), function()
+		SendRenAirstrike(airGroup5, airGroup5Route)
+	end)
+	Trigger.AfterDelay(DateTime.Seconds(720), SendAirstrike)
+	Trigger.AfterDelay(DateTime.Minutes(45), BuildBase)
```

Comment:
My guess is this was a proximity trigger but I'd have to check what the OG did.

Reply:
All timer which based on the original, multiplied by .6